### PR TITLE
(PC-17734) chore(eslint): add lint rule to avoid using single quote instead of apostrophe in strings

### PR DESCRIPTION
**Pull request by dbenfouzari-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-17734

## Checklist

I have:

- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](1)

## Screenshots

**delete** *if no UI change*

| Platform         | Before | After |
| :--------------- | :----: | :---: |
| iOS              |        |       |
| Android          |        |       |
| Phone - Chrome   |        |       |
| Tablet - Chrome  |        |       |
| Desktop - Chrome |        |       |
| Desktop - Safari |        |       |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -956,7 +956,7 @@ SPEC CHECKSUMS:
   FirebaseCore: 5743c5785c074a794d35f2fff7ecc254a91e08b1
   FirebaseCoreDiagnostics: 92e07a649aeb66352b319d43bdd2ee3942af84cb
   FirebaseDynamicLinks: 1dc816ef789c5adac6fede0b46d11478175c70e4
-  FirebaseFirestore: cb361b7f8f225a225c9f11b8d42066baebb1630c
```
Comment (Lucasbeneston) [inline ios/Podfile.lock:959]: Il faut vraiment commit ce changement ?

File: src/features/identityCheck/components/modals/SomeAdviceBeforeIdentityCheckModal.native.test.tsx [hunk 2]
```diff
@@ -31,7 +31,7 @@ describe('<SomeAdviceBeforeIdentityCheckModal/>', () => {
     expect(hideModalMock).toHaveBeenCalledTimes(1)
   })
 
-  it(`should call onPressContinue when cliking on "J'ai compris" button`, () => {
+  it('should call onPressContinue when cliking on "J\'ai compris" button', () => {
```
Comment (Lucasbeneston) [inline src/features/identityCheck/components/modals/SomeAdviceBeforeIdentityCheckModal.native.test.tsx:34]:
```suggestion
  it('should call onPressContinue when cliking on "J’ai compris" button', () => {
```

File: src/features/identityCheck/pages/identification/identificationStart/SelectPhoneStatus.web.test.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ describe('SelectPhoneStatus', () => {
     expect(navigate).toHaveBeenCalledWith('SelectIDStatus', undefined)
   })
 
-  it(`should navigate to DMSIntroduction FR when pressing "Je n'ai pas de smartphone" button`, () => {
+  it('should navigate to DMSIntroduction FR when pressing "Je n\'ai pas de smartphone" button', () => {
```
Comment (Lucasbeneston) [inline src/features/identityCheck/pages/identification/identificationStart/SelectPhoneStatus.web.test.tsx:22]:
```suggestion
  it('should navigate to DMSIntroduction FR when pressing "Je n’ai pas de smartphone" button', () => {
```

**Review by Lucasbeneston — APPROVED** [review]
Trop bien cette règle ESLint ! 🥇 🚀

File: ios/Podfile.lock (hunk 1 above)
Comment (dbenfouzari-pass) [reply ios/Podfile.lock:959]: Non tu as raison ! Je l'enlève.

**Review by Lucasbeneston — APPROVED** [review]

**Pull request merged**